neo/Ledger/MemPool.cs

```diff
@@ -0,0 +1,125 @@
+﻿using Neo.Network.P2P.Payloads;
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+
+namespace Neo.Ledger
+{
+    internal class MemPool : IReadOnlyCollection<Transaction>
+    {
+        private class PoolItem
+        {
+            public readonly Transaction Transaction;
+            public readonly DateTime Timestamp;
+
+            public PoolItem(Transaction tx)
+            {
+                Transaction = tx;
+                Timestamp = DateTime.UtcNow;
+            }
+        }
+
+        private const int MemoryPoolSize = 50000;
+        private readonly ConcurrentDictionary<UInt256, PoolItem> mem_pool = new ConcurrentDictionary<UInt256, PoolItem>();
+
+        public int Count => mem_pool.Count;
+
+        public void Clear()
+        {
+            mem_pool.Clear();
+        }
+
+        public bool ContainsKey(UInt256 hash)
+        {
+            return mem_pool.ContainsKey(hash);
+        }
+
+        public IEnumerator<Transaction> GetEnumerator()
+        {
+            return mem_pool.Select(p => p.Value.Transaction).GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        public void RemoveLowestFee(int count)
+        {
+            if (count <= 0) return;
+            if (count >= mem_pool.Count)
+            {
+                mem_pool.Clear();
+            }
+            else
+            {
+                UInt256[] delete = mem_pool.AsParallel()
+                    .OrderBy(p => p.Value.Transaction.NetworkFee / p.Value.Transaction.Size)
+                    .ThenBy(p => p.Value.Transaction.NetworkFee)
+                    .ThenBy(p => new BigInteger(p.Key.ToArray()))
+                    .Take(count)

```

Comment:
How well does the compiler optimize this if this is running to remove 1 transaction repeatedly?